app/models/amp/Amp.scala

```diff
@@ -16,63 +16,124 @@
+                     data: JsObject = Json.obj(),
+                     lastUpdated: LocalDateTime = LocalDateTime.now,
+                     hasChanged: Boolean = false,
+                     hasAccepted: Boolean = false) {
 
-  //TODO: Implement the methods based on AMP frontend once confirmed and ready to sync
+  /**
+    * Provides a means of setting data that will update the hasChanged flag
+    *
+    * Set data via this method and NOT directly in the constructor
+    */
+  def data(p: JsObject): Amp =
+    this.copy(data = p, hasChanged = hasChanged || this.data != p, hasAccepted = hasAccepted && this.data == p)
 
-  def BoughtOrSoldOverThreshold(): Amp  = ???
-  def DateTransactionOverThreshold(): Amp   = ???
-  def IdentifyLinkedTransactionsView(): Amp   = ???
-  def PercentageExpectedTurnover(): Amp   = ???
-  def TypeOfParticipant(): Amp   = ???
+  val typeOfParticipant            = JsPath \ "typeOfParticipant"
+  val typeOfParticipantDetail      = JsPath \ "typeOfParticipantDetail"
+  val boughtOrSoldOverThreshold    = JsPath \ "boughtOrSoldOverThreshold"
+  val identifyLinkedTransactions   = JsPath \ "identifyLinkedTransactions"
+  val dateTransactionOverThreshold = JsPath \ "dateTransactionOverThreshold"
+  val percentageExpectedTurnover   = JsPath \ "percentageExpectedTurnover"
+  val otherTypeOfParticipant       = "somethingelse"
 
-  def isComplete: Boolean = this match {
-    case Amp(true, accepted) => accepted
-    case _ => false
+  private def get[A](path: JsPath)(implicit rds: Reads[A]): Option[A] =
+    Reads.optionNoError(Reads.at(path)).reads(data).getOrElse(None)
+
+  private def isDefinedAt(path: JsPath): Boolean = {
+    get[JsValue](path).isDefined
+  }
+
+  private def valueAt(path: JsPath): String = {
+    get[JsValue](path).getOrElse("").toString().toLowerCase()
+  }
+
+  private def isTypeOfParticipantComplete: Boolean = {
+    isDefinedAt(typeOfParticipant) &&
+      ((valueAt(typeOfParticipant).contains(otherTypeOfParticipant) &&
+        isDefinedAt(typeOfParticipantDetail)) ||
+        (!valueAt(typeOfParticipant).contains(otherTypeOfParticipant)))
+  }
+
+  private def isBoughtOrSoldOverThresholdComplete: Boolean = {
+    isDefinedAt(boughtOrSoldOverThreshold) &&
+      ((valueAt(boughtOrSoldOverThreshold) == "true" &&
+        isDefinedAt(dateTransactionOverThreshold)) ||
+        (valueAt(boughtOrSoldOverThreshold) == "false"))
+  }
+
+  def isComplete: Boolean = {
+    isTypeOfParticipantComplete &&
+    isBoughtOrSoldOverThresholdComplete &&
+    isDefinedAt(identifyLinkedTransactions) &&
+    isDefinedAt(percentageExpectedTurnover)
   }
 }
 
-object Amp {
+object Amp extends ServicesConfig {
 
-  import play.api.libs.functional.syntax._
-  import play.api.libs.json._
+  val redirectCallType       = "GET"
+  val key                    = "amp"
+  lazy val ampWhatYouNeedUrl = s"${baseUrl("amls-art-market-participant-frontend")}amls-art-market-participant-frontend/what-you-need"
```

Comment:
Missing  "/" between "${baseUrl("amls-art-market-participant-frontend")}" and  "amls-art-market-participant-frontend/what-you-need"